packages/grafana-data/src/field/displayProcessor.ts

```diff
@@ -1,22 +1,18 @@
 // Libraries
 import _ from 'lodash';
 
-// Utils
-import { getColorFromHexRgbOrName } from '../utils/namedColorsPalette';
-
 // Types
-import { FieldConfig, FieldType } from '../types/dataFrame';
-import { GrafanaTheme, GrafanaThemeType } from '../types/theme';
+import { Field, FieldType } from '../types/dataFrame';
+import { GrafanaTheme } from '../types/theme';
 import { DisplayProcessor, DisplayValue, DecimalCount, DecimalInfo } from '../types/displayValue';
 import { getValueFormat } from '../valueFormats/valueFormats';
 import { getMappedValue } from '../utils/valueMappings';
-import { Threshold } from '../types/threshold';
 import { DEFAULT_DATE_TIME_FORMAT } from '../datetime';
 import { KeyValue } from '../types';
+import { getScaleCalculator } from '../utils';
 
 interface DisplayProcessorOptions {
-  type?: FieldType;
-  config?: FieldConfig;
+  field: Partial<Field>;
```

Comment:
@dprokop -- I know we discussed this a few times and were never sure when/why we may want access to the whole field.

For 'scale' it needs access to the possible range of data in the response, so I make this a partial field